src/amo/components/RatingsByStar/index.js

```diff
@@ -54,13 +54,25 @@ export class RatingsByStarBase extends React.Component<InternalProps> {
     const { addon, i18n, location } = this.props;
     const loading = !addon;
 
-    const linkTitles = {
+    const getLinkTitles = (rating, total) => {
+      const ratings = {
+        '5': 'five',
+        '4': 'four',
+        '3': 'three',
+        '2': 'two',
+        '1': 'one',
+      };
+
+      const countText = total && total > 1 ? `all ${total}` : 'the';
+      const reviewText = total && total > 1 ? 'reviews' : 'review';
+
       /* eslint-disable quote-props */
-      '5': i18n.gettext('Read all five-star reviews'),
-      '4': i18n.gettext('Read all four-star reviews'),
-      '3': i18n.gettext('Read all three-star reviews'),
-      '2': i18n.gettext('Read all two-star reviews'),
-      '1': i18n.gettext('Read all one-star reviews'),
+      if (total && total > 0) {
+        return i18n.gettext(
+          `Read ${countText} ${ratings[rating]}-star ${reviewText}`,
```

Comment:
This won't work with translations:
1. The string variables will be interpolated before being passed to `gettext()`
2. The dynamic bits (`all`/`the`/`review`/`reviews`) should be locale dependent. Some languages deal with pluralization differently, so you can't make the decision for what string to use based on English - you have to provide the variants and some information about what the number of items are, and let translators deal with it through `ngettext` (there should be a few uses of that function in the codebase that you can look at for inspiration).
3. Because of how dynamic this is this won't extract properly in the first place (you can see that by running `yarn extract-locales` in the branch - but don't commit the `locale/` changes if you do), so translators will never pick up the string to translate

See https://github.com/mozilla/addons-frontend/blob/master/docs/i18n.md#using-sprintf for handling of translated strings with variables and http://messageformat.github.io/Jed/ for more info about the various functions available. The gist of it is: avoid template strings, but also ensure translators will be able to translate all variants correctly. This is why there were 5 variants of the string before, it was easier to handle that way.